### PR TITLE
Reworked call view’s remote video mode.

### PR DESCRIPTION
**Pull request by charlesmchen**
* Hide speakerphone button if local video is enabled.
* Auto-enable speakerphone if local video is enabled.
* Combine button rows on call screen.
* Move mute and "toggle video" buttons down by "hang up" button if remote video is present.
* Various cleanup and bug fixes.

PTAL @michaelkirk 

// FREEBIE

img_6443
img_6442
img_6441
img_6440
img_6439

### CODE REVIEW

File: Signal/src/call/UserInterface/CallUIAdapter.swift [hunk 1]
```diff
@@ -145,15 +145,17 @@ extension CallUIAdaptee {
...
 
-    internal func toggleSpeakerphone(call: SignalCall, isEnabled: Bool) {
+    internal func setIsSpeakerphoneEnabled(call: SignalCall, isEnabled: Bool) {
         // Speakerphone is not handled by CallKit (e.g. there is no CXAction), so we handle it w/o going through the 
         // adaptee, relying on the AudioService CallObserver to put the system in a state consistent with the call's 
         // assigned property.
-        call.isSpeakerphoneEnabled = isEnabled
+        CallService.signalingQueue.async {
+            call.isSpeakerphoneEnabled = isEnabled
+        }
```
Comment (charlesmchen) [inline Signal/src/call/UserInterface/CallUIAdapter.swift]: This is a good example of one of the reasons I'm uncomfortable with how signaling queue is being used - it's easy to forget that CallService and SignalCall state should only be modified by the signaling queue.

File: Signal/src/view controllers/CallViewController.swift [hunk 2]
```diff
@@ -46,10 +46,15 @@ class CallViewController: UIViewController, CallObserver, CallServiceObserver, R
     var ongoingCallView: UIView!
 
     var hangUpButton: UIButton!
-    var muteButton: UIButton!
     var speakerPhoneButton: UIButton!
-    var textMessageButton: UIButton!
-    var videoButton: UIButton!
+    var audioMuteButton: UIButton!
+    var audioVideoButton: UIButton!
```
Comment (michaelkirk) [inline Signal/src/view controllers/CallViewController.swift]: These names confused me for 8 seconds.

File: Signal/src/view controllers/CallViewController.swift [hunk 3]
```diff
@@ -595,7 +600,7 @@ class CallViewController: UIViewController, CallObserver, CallServiceObserver, R
         Logger.info("\(TAG) called \(#function)")
         speakerphoneButton.isSelected = !speakerphoneButton.isSelected
         if let call = self.call {
-            callUIAdapter.toggleSpeakerphone(call: call, isEnabled: speakerphoneButton.isSelected)
+            callUIAdapter.setIsSpeakerphoneEnabled(call: call, isEnabled: speakerphoneButton.isSelected)
```
Comment (michaelkirk) [inline Signal/src/view controllers/CallViewController.swift]: 👍

File: Signal/src/view controllers/CallViewController.swift (hunk 2 above)
Comment (charlesmchen) [inline Signal/src/view controllers/CallViewController.swift]: Yes these names are terrible. I've renamed them.

File: Signal/src/view controllers/CallViewController.swift [hunk 4]
```diff
@@ -47,10 +47,10 @@ class CallViewController: UIViewController, CallObserver, CallServiceObserver, R
 
     var hangUpButton: UIButton!
     var speakerPhoneButton: UIButton!
-    var audioMuteButton: UIButton!
-    var audioVideoButton: UIButton!
-    var videoMuteButton: UIButton!
-    var videoVideoButton: UIButton!
+    var audioModeMuteButton: UIButton!
```
Comment (michaelkirk) [inline Signal/src/view controllers/CallViewController.swift]: 👏 👏 👏

**Pull request merged**